src/main/java/com/iota/iri/network/protocol/message/impl/MessageReaderImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.iota.iri.network.protocol.message.impl;
+
+import com.iota.iri.network.protocol.Protocol;
+import com.iota.iri.network.protocol.ProtocolMessage;
+import com.iota.iri.network.protocol.message.MessageReader;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+
+/**
+ * A {@link MessageReaderImpl} is a {@link MessageReader}.
+ */
+public class MessageReaderImpl implements MessageReader {
```

Comment:
Hmm in the Styleguide (https://github.com/iotaledger/iri/blob/dev/STYLEGUIDE.md#dont-document-overriding-methods-usually) it was recommended but I didn't make this mandatory for methods. The reason is that I don't want to chase people on that, and it is hard to automate.

By the way, in this case `@inheritDoc` doesn't work because it is a class that implements an interface.

If it was me I would write a comment like "Default implementation" (just to escape codacy). If there are interesting details to write about the implementation I would add them.

I wouldn't put `{@link MessageReaderImpl}` because we know what class we are on. I wouldn't put `{@link MessageReader}` because we get that link for free.

I am not insisting on my recommendation because I think I write enough comments when I review.